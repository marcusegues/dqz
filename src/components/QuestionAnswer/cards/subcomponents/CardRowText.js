import React from 'react';
import { Text } from 'react-native';
import { moderateScale, verticalScale } from '../../../../styles/Scaling';

const CardRowText = ({ text }) => (
  <Text
    style={{
      fontFamily: 'roboto_medium',
      fontSize: moderateScale(14),
      color: '#24253D',
    }}
  >
    {text}
  </Text>
);

export default CardRowText;
