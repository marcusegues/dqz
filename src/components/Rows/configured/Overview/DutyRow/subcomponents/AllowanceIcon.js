// @flow
import React from 'react';
// $FlowFixMe
import { View, Text, StyleSheet } from 'react-native';
import { moderateScale } from '../../../../../../styles/Scaling';
import { MAIN_BLACK } from '../../../../../../styles/colors';
import { type } from '../../../../../../styles/fonts';

const ownStyles = StyleSheet.create({
  quantityContainer: {
    marginTop: 5,
    paddingHorizontal: 6,
    paddingVertical: 2,
    backgroundColor: '#EBF8DD',
  },
  quantityText: {
    color: MAIN_BLACK,
    fontSize: moderateScale(12),
    fontFamily: type.light,
  },
});

type AllowanceIconProps = {
  quantity: number | string,
  unit: string,
  text: string,
};

export const AllowanceIcon = ({ quantity, unit, text }: AllowanceIconProps) => (
  <View style={ownStyles.quantityContainer}>
    <Text style={ownStyles.quantityText}>{`${text} ${quantity} ${unit}`}</Text>
  </View>
);
