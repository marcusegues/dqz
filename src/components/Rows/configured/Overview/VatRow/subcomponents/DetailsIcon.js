// @flow
import React from 'react';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';
import { Entypo } from '@expo/vector-icons';
// $FlowFixMe
import { View, Text, StyleSheet } from 'react-native';
import { moderateScale } from '../../../../../../styles/Scaling';

import type { TFunction } from '../../../../../../types/generalTypes';

const ownStyles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 5,
  },
  quantityText: {
    color: '#757575',
    fontSize: moderateScale(12),
    fontFamily: 'roboto_bold',
  },
});

type DetailsIconProps = {
  expanded: boolean,
};

export const DetailsIconInner = ({
  expanded,
  t,
}: DetailsIconProps & { t: TFunction }) => (
  <View style={ownStyles.container}>
    <View style={{ marginLeft: -5 }}>
      {expanded ? (
        <Entypo
          name="chevron-small-down"
          size={moderateScale(20)}
          color="#757575"
        />
      ) : (
        <Entypo
          name="chevron-small-right"
          size={moderateScale(20)}
          color="#757575"
        />
      )}
    </View>
    <Text style={ownStyles.quantityText}>{t('details').toUpperCase()}</Text>
  </View>
);

export const DetailsIcon = (translate(['overview'])(
  DetailsIconInner
): ComponentType<DetailsIconProps>);
