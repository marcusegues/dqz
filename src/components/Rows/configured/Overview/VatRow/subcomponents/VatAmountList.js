import React from 'react';
// $FlowFixMe
import { Text, View } from 'react-native';
import { translate } from 'react-i18next';
import { Row } from '../../../../Row';

class VatAmountListInner extends React.Component {
  render() {
    return (
      <Row
        borderBottom={false}
        onLayout={event => {
          console.log('Set max height', event.nativeEvent.layout.height);
          this.props.setContentHeight(event.nativeEvent.layout.height);
        }}
      >
        <Text>{'Attempt'}</Text>
      </Row>
    );
  }
}

export const VatAmountList = (translate(['receipt'])(
  VatAmountListInner
): ComponentType<VatRowProps>);
