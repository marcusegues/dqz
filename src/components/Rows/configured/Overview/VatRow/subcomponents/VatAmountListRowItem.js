// @flow
import React from 'react';
import type { ComponentType } from 'react';
import { connect } from 'react-redux';
// $FlowFixMe
import { View } from 'react-native';
import type { FlatAmount } from '../../../../../../model/utils';
import { rowStyles } from '../../../../styles/rowStyles';
import { Row } from '../../../../Row';
import { CardRowSubText } from '../../../../../QuestionAnswer/cards/subcomponents/CardRowSubText';
import { getCurrencies } from '../../../../../../reducers';
import type { CurrencyObject } from '../../../../../../model/currencies';
import { currencyPicker } from '../../../../../Modals/CurrencyPickerModal/currencyPickerData';
import { FlagAndCurrencyAmount } from '../../../../../General Components/FlagAndCurrencyAmount';

type VatAmountListRowItemProps = {
  flatAmount: FlatAmount,
  fullWidth?: boolean,
  borderTop?: boolean,
  borderBottom?: boolean,
};

type ReduxInject = {
  currencyObject: CurrencyObject,
};

const VatAmountListRowItemInner = ({
  flatAmount,
  fullWidth = false,
  borderTop = false,
  borderBottom = false,
  currencyObject,
}: VatAmountListRowItemProps & ReduxInject) => {
  const { currency, amount } = flatAmount;
  const flag = currencyPicker
    .filter(c => c.value === currency)
    .map(c => c.flag);

  return (
    <Row
      borderTop={borderTop}
      borderBottom={borderBottom}
      width={fullWidth ? '100%' : '95%'}
    >
      <View style={rowStyles.rowContent}>
        <FlagAndCurrencyAmount
          flag={flag}
          currency={currency}
          amount={amount}
        />
        <View style={{ flex: 1 }}>
          <CardRowSubText
            text={`~CHF ${(currencyObject[currency] * amount).toFixed(2)}`}
          />
        </View>
      </View>
    </Row>
  );
};

const mapStateToProps = state => ({
  currencyObject: getCurrencies(state),
});

export const VatAmountListRowItem = (connect(mapStateToProps)(
  VatAmountListRowItemInner
): ComponentType<VatAmountListRowItemProps>);
