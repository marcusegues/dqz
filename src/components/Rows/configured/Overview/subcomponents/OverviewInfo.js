// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { CardRowText } from '../../../../QuestionAnswer/cards/subcomponents/CardRowText';
import { CardRowSubText } from '../../../../QuestionAnswer/cards/subcomponents/CardRowSubText';

type OverviewInfoProps = {
  title: string,
  subtitle: string,
  children: any,
};

export const OverviewInfo = ({
  title,
  subtitle,
  children,
}: OverviewInfoProps) => {
  return (
    <View
      style={{
        flex: 1,
        flexDirection: 'column',
        alignItems: 'flex-start',
        paddingRight: 5,
      }}
    >
      <CardRowText text={title} />
      <CardRowSubText text={subtitle} />
      {children}
    </View>
  );
};
