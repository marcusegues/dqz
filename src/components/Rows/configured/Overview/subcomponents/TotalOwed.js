// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { CardRowText } from '../../../../QuestionAnswer/cards/subcomponents/CardRowText';

type TotalOwedProps = {
  result: string,
};

export const TotalOwed = ({ result }: TotalOwedProps) => (
  <View style={{ width: 70, flexDirection: 'row', justifyContent: 'flex-end' }}>
    <CardRowText text={result} />
  </View>
);
