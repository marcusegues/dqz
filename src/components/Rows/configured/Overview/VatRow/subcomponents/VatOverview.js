// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import type { ComponentType } from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { translate } from 'react-i18next';
import { OverviewInfo } from '../../subcomponents/OverviewInfo';
import { TotalOwed } from '../../subcomponents/TotalOwed';
import { AmountIcon } from '../../../../../General Components/GreyBox/configured/AmountIcon';
import { Row } from '../../../../Row';
import { rowStyles } from '../../../../styles/rowStyles';
import { DetailsIcon } from './DetailsIcon';
import type { TFunction } from '../../../../../../types/generalTypes';
import {
  totalLargeAmounts,
  totalNormalAmounts,
} from '../../../../../../model/utils';
import type { CurrencyObject } from '../../../../../../model/currencies';
import type { Amounts } from '../../../../../../model/types/basketPeopleAmountsTypes';

type VatOverviewProps = {
  large: boolean,
  amounts: Amounts,
  currencies: CurrencyObject,
  vat: number,
};

type CollapsibleInjected = {
  setMainHeight: (height: number) => void,
  animate: () => void,
  expanded: boolean,
};

const VatOverviewInner = ({
  large,
  amounts,
  currencies,
  vat,
  t,
  setMainHeight,
  animate,
  expanded,
}: VatOverviewProps & CollapsibleInjected & { t: TFunction }) => {
  const totalAmount = large
    ? totalLargeAmounts(amounts, currencies)
    : totalNormalAmounts(amounts, currencies);
  const title = large ? t('largeAmountsTitle') : t('normalAmountsTitle');
  const subTitle = large
    ? t('largeAmountsSubtitle')
    : t('normalAmountsSubtitle');
  return (
    <Row
      borderTop={false}
      borderBottom={false}
      width="100%"
      onLayout={event => {
        setMainHeight(event.nativeEvent.layout.height);
      }}
    >
      <View style={[rowStyles.rowContent]}>
        <OverviewInfo title={title} subtitle={subTitle}>
          <Touchable onPress={() => animate()}>
            <DetailsIcon expanded={expanded} />
          </Touchable>
        </OverviewInfo>
        <AmountIcon amount={totalAmount} currency="CHF" />
        {large ? (
          <View style={{ width: 70 }} />
        ) : (
          <TotalOwed result={(+vat).toFixed(2)} />
        )}
      </View>
    </Row>
  );
};

export const VatOverview = (translate(['receipt'])(
  VatOverviewInner
): ComponentType<VatOverviewProps>);
