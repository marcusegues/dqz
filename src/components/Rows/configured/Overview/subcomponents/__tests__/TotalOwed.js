// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { TotalOwed } from '../TotalOwed';
import { CardRowText } from '../../../../../QuestionAnswer/Cards/subcomponents/CardRowText';

describe('TotalOwed', () => {
  test('TotalOwed renders correctly', () => {
    const tree = renderer.create(<TotalOwed result="test result" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('shallow renders correctly and text matches', () => {
    const wrapper = shallow(<TotalOwed result="test result" />);
    const render = wrapper.dive();
    expect(render.find(CardRowText).props().text).toEqual('test result');
    expect(render).toMatchSnapshot();
  });
});
