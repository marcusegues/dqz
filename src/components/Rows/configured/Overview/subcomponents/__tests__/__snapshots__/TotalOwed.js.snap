// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalOwed TotalOwed renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "width": 70,
    }
  }
>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#24253D",
          "fontFamily": "roboto_medium",
          "fontSize": 22,
        }
      }
    >
      test result
    </Text>
  </View>
</View>
`;

exports[`TotalOwed shallow renders correctly and text matches 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "width": 70,
      }
    }
  >
    <CardRowText
      greyBox={false}
      style={Object {}}
      text="test result"
    />
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <CardRowText
        greyBox={false}
        style={Object {}}
        text="test result"
      />,
      "style": Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "width": 70,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "greyBox": false,
        "style": Object {},
        "text": "test result",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CardRowText
          greyBox={false}
          style={Object {}}
          text="test result"
        />,
        "style": Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": 70,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "test result",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
